Firestore/core/src/firebase/firestore/util/assert_apple.mm

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/util/assert.h"
+
+#import <Foundation/Foundation.h>
+
+#include <string>
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+namespace {
+
+// Translates a C format string to the equivalent NSString without making a
+// copy.
+NSString* CStringToNSString(const char* format) {
```

Comment:
This seems like a function we're going to use a lot when interoperating with platform native APIs on iOS.

Let's put this in a separate utility. As a separate utility we should make it clear that this is merely wrapping the C string rather than copying it. Maybe `WrapNSStringNoCopy`?